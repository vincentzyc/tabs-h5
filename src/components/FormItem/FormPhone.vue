<template>
  <div class="flex align-middle form-item">
    <div class="form-label">联系电话</div>
    <div class="form-input flex-auto flex align-middle">
      <input
        @blur="checkPhone(model)"
        :maxlength="11"
        placeholder="输入有效联系电话并保持畅通"
        type="tel"
        v-model.trim="model"
      />
    </div>
  </div>
</template>

<script lang="ts">
import { checkPhone } from "@/composition/business/use-verify-data";

export default defineComponent({
  name: "FormPhone",
  props: {
    modelValue: {
      required: true,
      type: String,
    },
  },
  emits: ["update:modelValue"],
  setup(props, { emit }) {
    const model = computed<string>({
      get() {
        return props.modelValue;
      },
      set(val) {
        emit("update:modelValue", val);
      },
    });
    return {
      model,
      checkPhone,
    };
  },
});
</script>
