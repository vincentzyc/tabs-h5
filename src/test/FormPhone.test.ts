import { mount } from '@vue/test-utils'
import FormPhone from '@/components/FormItem/FormPhone.vue'

test('mount FormPhone component', async () => {
  expect(FormPhone).toBeTruthy()

  const wrapper = mount(FormPhone, {
    props: {
      modelValue: '',
      'onUpdate:modelValue': (e: string) => wrapper.setProps({ modelValue: e })
    }
  })

  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.get('input').setValue('18812345678')

  expect(wrapper.props('modelValue')).toBe('18812345678')

  const emitEvent = wrapper.emitted('update:modelValue')

  expect(emitEvent).toHaveLength(1)
})
