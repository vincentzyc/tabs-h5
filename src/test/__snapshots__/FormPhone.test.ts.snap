// Vitest Snapshot v1

exports[`mount FormPhone component 1`] = `
"<div class=\\"flex align-middle form-item\\">
  <div class=\\"form-label\\">联系电话</div>
  <div class=\\"form-input flex-auto flex align-middle\\"><input maxlength=\\"11\\" placeholder=\\"输入有效联系电话并保持畅通\\" type=\\"tel\\"></div>
</div>"
`;
